webdriver/tests/bidi/script/call_function/sandbox.py

```diff
@@ -0,0 +1,153 @@
+import pytest
+
+from webdriver.bidi.modules.script import ContextTarget, ScriptEvaluateResultException
+
+from ... import any_int, any_string, recursive_compare
+from .. import any_stack_trace
+
+
+@pytest.mark.asyncio
+async def test_sandbox(bidi_session, new_tab):
+    # Make changes in window
+    await bidi_session.script.call_function(
+        function_declaration="() => { window.foo = 1; }",
+        target=ContextTarget(new_tab["context"]),
+        await_promise=True,
+    )
+
+    # Check that changes are not present in sandbox
+    result_in_sandbox = await bidi_session.script.call_function(
+        function_declaration="() => window.foo",
+        target=ContextTarget(new_tab["context"], "sandbox"),
+        await_promise=True,
+    )
+    assert result_in_sandbox == {"type": "undefined"}
+
+    # Make changes in sandbox
+    await bidi_session.script.call_function(
+        function_declaration="() => { window.bar = 2; }",
+        target=ContextTarget(new_tab["context"], "sandbox"),
+        await_promise=True,
+    )
+
+    # Make sure that changes are present in sandbox
+    result_in_sandbox = await bidi_session.script.call_function(
+        function_declaration="() => window.bar",
+        target=ContextTarget(new_tab["context"], "sandbox"),
+        await_promise=True,
+    )
+    assert result_in_sandbox == {"type": "number", "value": 2}
+
+    # Make sure that changes didn't leak from sandbox
+    result_in_window = await bidi_session.script.call_function(
+        function_declaration="() => window.bar",
+        target=ContextTarget(new_tab["context"]),
+        await_promise=True,
+    )
+    assert result_in_window == {"type": "undefined"}
+
+
+@pytest.mark.asyncio
+async def test_sandbox_with_empty_name(bidi_session, new_tab):
+    await bidi_session.script.call_function(
+        function_declaration="() => window.foo = 'bar'",
+        target=ContextTarget(new_tab["context"], ""),
+        await_promise=True,
+    )
+
+    # Make sure that we can find the sandbox with the empty name
+    result = await bidi_session.script.call_function(
+        function_declaration="() => window.foo",
+        target=ContextTarget(new_tab["context"], ""),
+        await_promise=True,
+    )
+    assert result == {"type": "string", "value": "bar"}
+
+    # Make sure that changes didn't leak from sandbox
+    result = await bidi_session.script.evaluate(
+        expression="window.foo",
+        target=ContextTarget(new_tab["context"]),
+        await_promise=True,
+    )
+    assert result == {"type": "undefined"}
+
+
+@pytest.mark.asyncio
+async def test_switch_sandboxes(bidi_session, new_tab):
```

Comment:
NIT: `test_sandbox`, `test_sandbox_with_empty_name` and `test_switch_sandboxes` can be a single parametrised test.

Reply:
I think I would rather keep it like this for now, so it's clearer which use cases we are testing